pyproject.toml

```diff
@@ -96,11 +96,11 @@ dev = [
     # Also update [build-system] -> requires
     'meson-python>=0.13',
     'wheel',
-    'setuptools>=67',
+    'setuptools~=69.5',
     'packaging>=21',
     'ninja',
     'Cython>=0.29.35',
-    'numpy>=1.22',
+    'numpy~=1.22',
```

Comment:
the update of pyproject.toml is not needed after the merge of #3274.

it should works with numpy 2.0.0 and below.